src/data/polynomial/eval.lean

```diff
@@ -527,6 +551,15 @@ by apply is_ring_hom.of_semiring
 
 instance eval.is_ring_hom {x : R} : is_ring_hom (eval x) := eval₂.is_ring_hom _
 
+lemma eval₂_endomorphism_algebra_map {M : Type w}
+  [add_comm_group M] [module R M]
+  (f : M →ₗ[R] M) (v : M) (p : polynomial R) :
+  p.eval₂ (algebra_map R (M →ₗ[R] M)) f v = p.sum (λ n b, b • (f ^ n) v) :=
```

Comment:
Unfortunately, `aeval` requires commutativity and linear maps are not commutative. There is a TODO note on this in `data/polynomial/algebra_map.lean`, but I think it will be a larger effort to change the definition.

Reply:
Makes sense, we can postpone that to a later PR then.